### PR TITLE
Refactor/documentation on various headwear

### PR DESCRIPTION
**Pull request by ivanmixo**
## About The Pull Request

- Refactors a lot of hat code, mostly old stuff. It sucks a little less now.
- Deletes unnecessary procs, unused items, useless vars
- Repaths caps a little
- Fixes formatting on some stuff like warden's drill hat, removes exclamation points from names of some hats
- Adds a bunch of comments

## Why It's Good For The Game

Cleaner code, happier coders.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

I did a lot of testing but no pwoof...

https://user-images.githubusercontent.com/22431091/159140502-447d3639-a238-450b-90f8-307f8fd0c0ca.mov

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

## Changelog
:cl:
add: Riot helmets can now actually have seclites attached
del: Removed a bunch of old useless procs/verbs/vars
fix: The light on the justice sec helmet now actually spins
refactor: Refactored a bunch of the headwear code, it sucks a tiny bit less now
/:cl:

### CODE REVIEW

**Comment by ivanmixo** [comment]
hot

File: code/modules/clothing/head/hardhat.dm [hunk 1]
```diff
@@ -1,15 +1,29 @@
...
 	icon_state = "hardhat0_yellow"
 	item_state = "hardhat0_yellow"
-	var/hat_type = "yellow" //Determines used sprites: hardhat[on]_[hat_type] and hardhat[on]_[hat_type]2 (lying down sprite)
 	armor = list("melee" = 15, "bullet" = 5, "laser" = 20, "energy" = 10, "bomb" = 20, "bio" = 10, "rad" = 20, "fire" = 100, "acid" = 50, "stamina" = 20)
-	flags_inv = 0
+	flags_inv = NONE
 	actions_types = list(/datum/action/item_action/toggle_helmet_light)
 	resistance_flags = FIRE_PROOF
 	clothing_flags = SNUG_FIT
-	dynamic_hair_suffix = "+generic"
```
Comment (ike709) [inline code/modules/clothing/head/hardhat.dm:12]: Why's this being removed?

**Review by ike709 — APPROVED** [review]
Reviewed almost everything except I ran out of time before finishing `code/modules/clothing/head/hardhat.dm` and `code/modules/clothing/head/helmet.dm` so I'll leave those for another maintainer (or revisit later if they don't).

Largely looks fine. It'd be nice if this also converted the softcap and hardhat to greyscale and deduplicated the icons, but this is still an improvement.

Good work 👍🏻

File: code/modules/clothing/head/hardhat.dm (hunk 1 above)
Comment (ivanmixo) [reply code/modules/clothing/head/hardhat.dm:12]: Because its already set to `+generic` on the parent

**Pull request merged**